app/assets/javascripts/oxalis/geometries/arbitrary_plane.js

```diff
@@ -122,75 +86,21 @@ class ArbitraryPlane {
       mesh.matrixWorldNeedsUpdate = true;
 
       this.isDirty = false;
-    }
-  }
-
-  calculateSphereVertices = _.memoize(sphericalCapRadius => {
-    const queryVertices = new Float32Array(this.width * this.width * 3);
-
-    // so we have Point [0, 0, 0] centered
-    let currentIndex = 0;
-
-    const vertex = [0, 0, 0];
-    let vector = [0, 0, 0];
-    const centerVertex = [0, 0, -sphericalCapRadius];
-
-    // Transforming those normalVertices to become a spherical cap
-    // which is better more smooth for querying.
-    // http://en.wikipedia.org/wiki/Spherical_cap
-    for (let y = 0; y < this.width; y++) {
-      for (let x = 0; x < this.width; x++) {
-        vertex[0] = x - Math.floor(this.width / 2);
-        vertex[1] = y - Math.floor(this.width / 2);
-        vertex[2] = 0;
-
-        vector = V3.sub(vertex, centerVertex, vector);
-        const length = V3.length(vector);
-        vector = V3.scale(vector, sphericalCapRadius / length, vector);
-
-        queryVertices[currentIndex++] = centerVertex[0] + vector[0];
-        queryVertices[currentIndex++] = centerVertex[1] + vector[1];
-        queryVertices[currentIndex++] = centerVertex[2] + vector[2];
-      }
-    }
-
-    return queryVertices;
-  });
-
-  calculatePlaneVertices = _.memoize(() => {
-    const queryVertices = new Float32Array(this.width * this.width * 3);
 
-    // so we have Point [0, 0, 0] centered
-    let currentIndex = 0;
-
-    for (let y = 0; y < this.width; y++) {
-      for (let x = 0; x < this.width; x++) {
-        queryVertices[currentIndex++] = x - Math.floor(this.width / 2);
-        queryVertices[currentIndex++] = y - Math.floor(this.width / 2);
-        queryVertices[currentIndex++] = 0;
-      }
-    }
-
-    return queryVertices;
-  });
-
-  setSphericalCapRadius(sphericalCapRadius: number) {
-    if (Store.getState().temporaryConfiguration.viewMode === constants.MODE_ARBITRARY) {
-      this.queryVertices = this.calculateSphereVertices(sphericalCapRadius);
-      this.isDirty = true;
+      SceneController.update(this);
     }
   }
 
   createMesh() {
-    this.textureMaterial = new ArbitraryPlaneMaterialFactory(this.width, {}).setup().getMaterial();
+    this.textureMaterial = new PlaneMaterialFactory(0, OrthoViews.PLANE_XY, 4)
```

Comment:
I just removed the width which was superfluous. PLANE_XY is used since all the dimension transpositions should act like in PLANE_XY. In the custom-viewports PR, I added another "arbitrary" viewport type, which I could use here. But I'd wait until that is merged into master.